src/registrar/templatetags/custom_filters.py

```diff
@@ -67,3 +67,66 @@ def get_organization_long_name(generic_org_type):
 @register.filter(name="has_permission")
 def has_permission(user, permission):
     return user.has_perm(permission)
+
+
+@register.filter
+def get_region(state):
+    regions = {
+        "CT": 1,
+        "ME": 1,
+        "MA": 1,
+        "NH": 1,
+        "RI": 1,
+        "VT": 1,
+        "NJ": 2,
+        "NY": 2,
+        "PR": 2,
+        "VI": 2,
+        "DE": 3,
+        "DC": 3,
+        "MD": 3,
+        "PA": 3,
+        "VA": 3,
+        "WV": 3,
+        "AL": 4,
+        "FL": 4,
+        "GA": 4,
+        "KY": 4,
+        "MS": 4,
+        "NC": 4,
+        "SC": 4,
+        "TN": 4,
+        "IL": 5,
+        "IN": 5,
+        "MI": 5,
+        "MN": 5,
+        "OH": 5,
+        "WI": 5,
+        "AR": 6,
+        "LA": 6,
+        "NM": 6,
+        "OK": 6,
+        "TX": 6,
+        "IA": 7,
+        "KS": 7,
+        "MO": 7,
+        "NE": 7,
+        "CO": 8,
+        "MT": 8,
+        "ND": 8,
+        "SD": 8,
+        "UT": 8,
+        "WY": 8,
+        "AZ": 9,
+        "CA": 9,
+        "HI": 9,
+        "NV": 9,
+        "GU": 9,
+        "AS": 9,
+        "MP": 9,
+        "AK": 10,
+        "ID": 10,
+        "OR": 10,
+        "WA": 10,
+    }
+    return regions.get(state.upper(), None)
```

Comment:
You're calling upper() but never check that this is not none (or if it's a string), so if this is every called where organization state/territory is empty it will get a 500 error. For proof go to hey12.gov (500 after I changed it to None to break stuff)
![image](https://github.com/cisagov/manage.get.gov/assets/109625347/ff0426d4-a2d9-471e-9ce5-79802ea64a5e)

Reply:
Good grief and good catch